pkg/config/config.go

```diff
@@ -564,3 +575,65 @@ func parseFeatureGates(featureGatesRaw string) (map[string]bool, error) {
 
 	return featureGatesMap, nil
 }
+
+// GetReconcileResources returns a slice of resource kinds that should be reconciled.
+func (cfg *Config) GetReconcileResources() ([]string, error) {
+	return parseReconcileResourcesString(cfg.ReconcileResources)
+}
+
+// parseReconcileResourcesString parses the reconcileResources flag and returns a slice
+// of resource kinds to reconcile.
+func parseReconcileResourcesString(resources string) ([]string, error) {
+	// reconcile all resources
+	if resources == "" {
+		return nil, nil
+	}
+
+	visited := make(map[string]bool)
+	resourceKinds := []string{}
+
+	for _, kind := range strings.Split(resources, ",") {
+		kind = strings.TrimSpace(kind)
+		if kind == "" {
+			return nil, fmt.Errorf("invalid resource kind: empty kind")
+		}
+		if _, ok := visited[kind]; ok {
+			return nil, fmt.Errorf("duplicate resource kind '%s'", kind)
+		}
+		visited[kind] = true
+		resourceKinds = append(resourceKinds, kind)
+	}
+	return resourceKinds, nil
+}
+
+// validateReconcileResources validates that the specified resource kinds are supported by the controller.
+func (cfg *Config) validateReconcileResources(supportedGVKs []schema.GroupVersionKind) error {
+	resources, err := cfg.GetReconcileResources()
+	if err != nil {
+		return fmt.Errorf("invalid value for flag '%s': %v", flagReconcileResources, err)
+	}
+	if resources == nil {
+		return nil
+	}
+
+	validResourceKinds := make(map[string]bool)
+	for _, gvk := range supportedGVKs {
+		validResourceKinds[gvk.Kind] = true
+	}
+
+	for _, resource := range resources {
+		if !validResourceKinds[resource] {
+			validKinds := make([]string, 0, len(validResourceKinds))
+			for kind := range validResourceKinds {
+				validKinds = append(validKinds, kind)
+			}
```

Comment:
this block is making a map of valid kinds, then making an array of valid kinds, can we reduce the complexity by just using an array and slice.Contains?